src/scale.jl

```diff
@@ -173,6 +174,9 @@ function apply_scale(scale::ContinuousScale,
     for (aes, data) in zip(aess, datas)
         for var in scale.vars
             vals = getfield(data, var)
+            if vals isa CategoricalArray
```

Comment:
Is there a strong reason to throw an error here? I fully agree that by default `CategoricalArray` should be treated as categorical even if it contains numeric values (that was the rationale for moving from `PooledDataArray`), but if the user explicitly requests a continuous scale, why not accept it? Or is it because the code fails later?

Reply:
> is there a strong reason to throw an error here?

Yes. It won't work. The `CategorialcalValue`s will cause failures sooner or later so I thought an informative error would be more helpful. The specific error in the example I tried is a missing `isfinite(::CategoricalValue)`. As mentioned previously, the core of this PR is to avoid that `CategoricalValue`s escape to `Compose`. In the discrete case, all data vectors go through `discretize` where `CategoricalArray`s are converted to `IndirectArrays` to avoid `CategoricalValue`s but in the continuous case there isn't a similar function we can use for making sure all `CategoricalArray`s are converted.

> that was the rationale for moving from `PooledDataArray`

I don't really understand why it wouldn't be exactly the same thing for `PooledDataArray`s. I'd also expect them to be treated as categorical.

Reply:
OK. That can always be improved later anyway if somebody needs it. Thanks for finishing this porting work!

> I don't really understand why it wouldn't be exactly the same thing for PooledDataArrays. I'd also expect them to be treated as categorical.

Yeah, but there had always been an ambiguity, as PDAs were also used as an efficient storage type (e.g. they supported math operations). Now the distinction should be clearer, if only because of the `CategoricalArray` name.